core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -1647,4 +1647,28 @@ class ParserTest extends munit.ScalaCheckSuite {
       assert(res.isLeft)
     }
   }
+
+  property("select on pure values works as expected") {
```

Comment:
somehow the comment on the PR "This is a law that shows that when first argument of the select is Right, we don't look at the second parser." was easier to understand to me than either "works as expected" or the code below.

Reply:
PR for such a change would be welcomed!